File: tuesday/morning/claude_exercise/figure2.c
#include <stdio.h>
#include <stdlib.h>

#define N_STEPS 150

int main() {
    double trajectory[N_STEPS + 1];
    double x0, x_current;
    int t;
    FILE *fp;

    // Initial condition
    x0 = 0.3;
    trajectory[0] = x0;
    x_current = x0;

    // Generate time series using quadratic map g(x) = 4x(1-x)
    for (t = 1; t <= N_STEPS; t++) {
        x_current = 4.0 * x_current * (1.0 - x_current);
        trajectory[t] = x_current;
    }

    // Write to file
    fp = fopen("figure2_data_c.txt", "w");
    if (fp == NULL) {
        fprintf(stderr, "Error opening file\n");
        return 1;
    }

    fprintf(fp, "# t x_t\n");
    for (t = 0; t <= N_STEPS; t++) {
        fprintf(fp, "%d %.10f\n", t, trajectory[t]);
    }
    fclose(fp);

    printf("Figure 2 data written to figure2_data_c.txt\n");

    return 0;
}

File: tuesday/morning/claude_exercise/figure2.f90
program figure2
    implicit none
    integer, parameter :: n_steps = 150
    real(8) :: x0, x_current
    real(8), dimension(0:n_steps) :: trajectory
    integer :: t

    ! Initial condition
    x0 = 0.3d0
    trajectory(0) = x0
    x_current = x0

    ! Generate time series using quadratic map g(x) = 4x(1-x)
    do t = 1, n_steps
        x_current = 4.0d0 * x_current * (1.0d0 - x_current)
        trajectory(t) = x_current
    end do

    ! Write to file
    open(unit=10, file='figure2_data.txt', status='replace')
    write(10, '(A)') '# t x_t'
    do t = 0, n_steps
        write(10, '(I10, F20.10)') t, trajectory(t)
    end do
    close(10)

    print *, 'Figure 2 data written to figure2_data.txt'

end program figure2
